cloudinit.net refactor: apply_network_config_names

aciba90 opened the pull request:
## Proposed Commit Message
<!-- Include a proposed commit message because all PRs are squash merged -->

```
cloudinit.net refactor: apply_network_config_names

- Remove `apply_network_config_names` from `cloudinit.net` and `Distro` classes so it is only accessible via the networking attribute.

LP: #1884602
```

## Checklist:
<!-- Go over all the following points, and put an `x` in all the boxes
that apply. -->
 - [x] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/topics/contributing.html)
 - [x] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any documentation accordingly

TheRealFalcon reviewed (commented):
@aciba90 , thanks for this contribution.

I think the end goal of the net refactor (which has admittedly stalled), is to remove methods from the distro classes entirely and into the networking classes. So the call in [stages.py](https://github.com/canonical/cloud-init/blob/e783ef5c8e5f7c46816161f6a902f2912a4cc93f/cloudinit/stages.py#L855) would be `self.distro.networking.apply_network_config_names(netcfg)` instead.

In order to do that, we would also need implementations in the BSD networking class. The current implementations (e.g., [1](https://github.com/canonical/cloud-init/blob/e783ef5c8e5f7c46816161f6a902f2912a4cc93f/cloudinit/distros/bsd.py#L138) [2](https://github.com/canonical/cloud-init/blob/e783ef5c8e5f7c46816161f6a902f2912a4cc93f/cloudinit/distros/freebsd.py#L156) [3](https://github.com/canonical/cloud-init/blob/e783ef5c8e5f7c46816161f6a902f2912a4cc93f/cloudinit/distros/netbsd.py#L137) ) are unsupported or no-ops and would also need to be refactored into this networking class setup.

aciba90 commented:
Thank you for your comments.

> I think the end goal of the net refactor (which has admittedly stalled), is to remove methods from the distro classes entirely and into the networking classes. So the call in [stages.py](https://github.com/canonical/cloud-init/blob/e783ef5c8e5f7c46816161f6a902f2912a4cc93f/cloudinit/stages.py#L855) would be `self.distro.networking.apply_network_config_names(netcfg)` instead.

I have removed those methods and made the change in `stages.py`.

> In order to do that, we would also need implementations in the BSD networking class. The current implementations (e.g., [1](https://github.com/canonical/cloud-init/blob/e783ef5c8e5f7c46816161f6a902f2912a4cc93f/cloudinit/distros/bsd.py#L138) [2](https://github.com/canonical/cloud-init/blob/e783ef5c8e5f7c46816161f6a902f2912a4cc93f/cloudinit/distros/freebsd.py#L156) [3](https://github.com/canonical/cloud-init/blob/e783ef5c8e5f7c46816161f6a902f2912a4cc93f/cloudinit/distros/netbsd.py#L137) ) are unsupported or no-ops and would also need to be refactored into this networking class setup.

I have moved current implementations into `BSDNetworking` and `FreeBSDNetworking`.

@TheRealFalcon please let me know if there is something else to improve.

TheRealFalcon commented on cloudinit/distros/freebsd.py:
```diff
@@ -13,16 +13,28 @@
 LOG = logging.getLogger(__name__)
 
 
+class FreeBSDNetworking(BSDNetworking):
```
Let's put this class in `networking.py`. For now, I think it makes more sense there.

aciba90 replied:
Done

TheRealFalcon commented on tests/unittests/distros/test_networking.py:
```diff
@@ -229,3 +235,113 @@ def test_retrying_and_strict_behaviour(
         )
+
+
+class TestLinuxNetworkingApplyNetworkCfgNames:
```
Nice refactor of these tests! This is much nicer.

aciba90 replied:
Thanks!

TheRealFalcon commented on tests/unittests/distros/test_networking.py:
```diff
@@ -229,3 +235,113 @@ def test_retrying_and_strict_behaviour(
         )
+
+
+class TestLinuxNetworkingApplyNetworkCfgNames:
```
Nice refactor of these tests! This is much nicer.

TheRealFalcon commented on tests/unittests/distros/test_networking.py:
```diff
@@ -229,3 +235,113 @@ def test_retrying_and_strict_behaviour(
+            pytest.param("V2_CONFIG_NO_MAC", id="without_mac"),
+        ],
+    )
+    def test_apply_v2_renames_skips_without_setname(self, config_attr: str):
```
Nit: `test_apply_v2_renames_skips_without_setname_or_mac`

aciba90 replied:
Done

TheRealFalcon commented on tests/unittests/distros/test_networking.py:
```diff
@@ -229,3 +235,113 @@ def test_retrying_and_strict_behaviour(
+            pytest.param("V2_CONFIG_NO_MAC", id="without_mac"),
+        ],
+    )
+    def test_apply_v2_renames_skips_without_setname(self, config_attr: str):
```
Nit: `test_apply_v2_renames_skips_without_setname_or_mac`

TheRealFalcon reviewed (changes requested):
Thanks for addressing the requested changes. One more minor change in where the new networking class lives, and I think we'll be good here.

aciba90 commented:
> Thanks for addressing the requested changes. One more minor change in where the new networking class lives, and I think we'll be good here.

Thanks for the comments. Resolved!

TheRealFalcon reviewed (approved):
LGTM, thanks! I'm going to briefly close and then reopen this PR to trigger a Travis rebuild.
